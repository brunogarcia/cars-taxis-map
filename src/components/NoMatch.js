import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import AlertIcon from '@material-ui/icons/Info';
import Typography from '@material-ui/core/Typography';

const NoMatch = ({ location }) => (
  <Fragment>
    <Typography variant="headline" component="h3">
      <AlertIcon /> Error
    </Typography>
    <Typography componentnt="p">
      No match for <em>{location.pathname}</em>
    </Typography>
  </Fragment>
);

NoMatch.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

export default NoMatch;
