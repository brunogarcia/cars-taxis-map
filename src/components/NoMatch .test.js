import React from 'react';
import renderer from 'react-test-renderer';
import NoMatch from './NoMatch';

test('renders correctly', () => {
  const props = {
    history: {
      push: () => {},
    },
    location: {
      pathname: '/path/to/somewhere',
    },
  };

  const component = renderer.create(<NoMatch {...props} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
