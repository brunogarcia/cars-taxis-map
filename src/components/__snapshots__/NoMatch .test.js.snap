// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h3
    className="MuiTypography-root-1 MuiTypography-headline-6"
>
    <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-25"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
    >
        <g>
            <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
        </g>
    </svg>
     Error
</h3>,
  <p
    className="MuiTypography-root-1 MuiTypography-body1-10"
    componentnt="p"
>
    No match for 
    <em>
        /path/to/somewhere
    </em>
</p>,
]
`;
